janus_core/calculations/geom_opt.py

```diff
@@ -187,13 +215,32 @@ def __init__(
         if log_kwargs and "filename" not in log_kwargs:
             raise ValueError("'filename' must be included in `log_kwargs`")
 
-        # Configure logging
+        # Read last image by default
+        read_kwargs.setdefault("index", -1)
+
+        # Set log name
         log_kwargs.setdefault("name", __name__)
-        self.logger = config_logger(**log_kwargs)
-        self.tracker = config_tracker(self.logger, **tracker_kwargs)
+
+        # Initialise structures and logging
+        super().__init__(
+            struct=struct,
+            struct_path=struct_path,
+            arch=arch,
+            device=device,
+            model_path=model_path,
+            read_kwargs=read_kwargs,
+            sequence_allowed=False,
+            calc_kwargs=calc_kwargs,
+            set_calc=set_calc,
+            log_kwargs=log_kwargs,
+            tracker_kwargs=tracker_kwargs,
+        )
+
+        if not self.struct.calc:
+            raise ValueError("Please attach a calculator to `struct`.")
 
         # Set output file
-        FileNameMixin.__init__(self, self.struct, None)
+        FileNameMixin.__init__(self, self.struct, self.struct_path, None)
```

Comment:
Will this move into the base calculation?

Reply:
There may be a nicer/more general way to pass additional, but implemented a potential fix, good idea